src/pyOpenMS/addons/MSExperiment.pyx

```diff
@@ -171,3 +173,341 @@
             parts.append(f"rt_range=[{min_rt:.2f}, {max_rt:.2f}]")
 
         return f"MSExperiment({', '.join(parts)})"
+
+    def get_df_columns(self, long_format=False):
+        """
+        get_df_columns(self: MSExperiment, long_format: bool = False) -> List[str]
+
+        Returns a list of column names that get_df() would produce.
+
+        Useful for discovering available columns before export.
+
+        Args:
+            long_format (bool): If True, returns columns for long format.
+                               If False, returns columns for compact format.
+
+        Returns:
+            list: List of column name strings.
+
+        Example:
+            >>> exp.get_df_columns(long_format=True)
+            ['rt', 'mz', 'intensity', 'ms_level']
+
+            >>> exp.get_df_columns(long_format=False)
+            ['rt', 'ms_level', 'mz_array', 'intensity_array']
+        """
+        if long_format:
+            return ['rt', 'mz', 'intensity', 'ms_level']
+        else:
+            return ['rt', 'ms_level', 'mz_array', 'intensity_array']
```

Comment:
Is it possible to include the scan number / native id? There was an instance when I wanted to extract data for a specific spectrum using the scan number, but the `MSExperiment.get_df` doesn't include this meta data. Doesn't need to be included in this PR, but was just wondering if that would be possible?